code/modules/mob/living/simple_animal/hostile/megafauna/drake.dm

```diff
@@ -263,71 +282,66 @@ Difficulty: Medium
-	SetRecoveryTime(MEGAFAUNA_DEFAULT_RECOVERY_TIME)
+	swoop_cooldown = world.time + cooldown
+	
+/mob/living/simple_animal/hostile/megafauna/dragon/proc/goto_target()
+	if(!target)
+		return
+	forceMove(get_turf(target))
 
 /mob/living/simple_animal/hostile/megafauna/dragon/AltClickOn(atom/movable/A)
 	if(!istype(A))
 		return
-	if(swoop_cooldown >= world.time)
+	if(player_cooldown >= world.time)
 		to_chat(src, "<span class='warning'>You need to wait 20 seconds between swoop attacks!</span>")
 		return
-	swoop_attack(TRUE, A, 25)
+	swoop_attack(TRUE, A, 30)
+	player_cooldown = world.time + 200
 
 /obj/item/gps/internal/dragon
 	icon_state = null
 	gpstag = "Fiery Signal"
 	desc = "Here there be dragons."
 	invisibility = 100
 
-
-/obj/effect/temp_visual/fireball
-	icon = 'icons/obj/wizard.dmi'
-	icon_state = "fireball"
-	name = "fireball"
-	desc = "Get out of the way!"
-	layer = FLY_LAYER
-	randomdir = FALSE
-	duration = 9
-	pixel_z = DRAKE_SWOOP_HEIGHT
-
-/obj/effect/temp_visual/fireball/Initialize()
-	. = ..()
-	animate(src, pixel_z = 0, time = duration)
-
 /obj/effect/temp_visual/target
-	icon = 'icons/mob/actions/actions_items.dmi'
-	icon_state = "sniper_zoom"
+	icon_state = "lavastaff_warn"
 	layer = BELOW_MOB_LAYER
 	light_range = 2
-	duration = 9
+	duration = 11
 
 /obj/effect/temp_visual/target/ex_act()
 	return
 
-/obj/effect/temp_visual/target/Initialize(mapload, list/flame_hit)
+/obj/effect/temp_visual/target/Initialize(mapload, var/reset_time = 10)
 	. = ..()
-	INVOKE_ASYNC(src, .proc/fall, flame_hit)
-
-/obj/effect/temp_visual/target/proc/fall(list/flame_hit)
+	INVOKE_ASYNC(src, .proc/fall, reset_time)
+	src.alpha = 127.5
+	animate(src, alpha = 255, time = duration)
+	
+/obj/effect/temp_visual/target/proc/fall(var/reset_time)
 	var/turf/T = get_turf(src)
 	playsound(T,'sound/magic/fleshtostone.ogg', 80, 1)
-	new /obj/effect/temp_visual/fireball(T)
 	sleep(duration)
-	if(ismineralturf(T))
-		var/turf/closed/mineral/M = T
-		M.gets_drilled()
-	playsound(T, "explosion", 80, 1)
-	new /obj/effect/hotspot(T)
-	T.hotspot_expose(700, 50, 1)
+	playsound(T,'sound/magic/fireball.ogg', 200, 1)
+	
 	for(var/mob/living/L in T.contents)
 		if(istype(L, /mob/living/simple_animal/hostile/megafauna/dragon))
 			continue
-		if(islist(flame_hit) && !flame_hit[L])
-			L.adjustFireLoss(40)
-			to_chat(L, "<span class='userdanger'>You're hit by the drake's fire breath!</span>")
-			flame_hit[L] = TRUE
-		else
-			L.adjustFireLoss(10) //if we've already hit them, do way less damage
+		L.adjustFireLoss(10)
+		to_chat(L, "<span class='userdanger'>You fall directly into the pool of lava!</span>")
+		
+	// deals damage to mechs
+	for(var/obj/mecha/M in T.contents)
+		M.take_damage(60, BRUTE, "melee", 1)
+		
+	// changes turf to lava temporarily
+	if(!istype(T, /turf/closed) && !istype(T, /turf/open/lava))
+		var/lava_turf = /turf/open/lava/smooth
+		var/reset_turf = T.type
+		T.ChangeTurf(lava_turf)
+		sleep(reset_time)
+		T.ChangeTurf(reset_turf)
```

Comment:
PlaceOnTop might be better if you intend to reset the turf later

Reply:
Not sure how I’d get rid of the turf on top afterwards, It works how I want it to though so I don’t really feel the need to change it unless it actually might be more efficient.

Reply:
It will cause issues if for example a drake casts lava on already lava-ed turfs, which means that they'll remember the lava as the last turf instead of the original turf, making it permanent. Even if one drake alone can't do that, it might happen if two drakes are close to each other.

Reply:
That's already in the code and it checks if the type if lava so that bug doesn't happen. There was a previous attack that could layer so that's why I made it.